src/ZPublisher/BaseRequest.py

```diff
@@ -451,6 +457,19 @@ def traverse(self, path, response=None, validated_hook=None):
                     # (usually self) and a sequence of names to traverse to
                     # find the method to be published.
 
+                    # This is webdav support. The last object in the path
+                    # should not be acquired. Instead, a NullResource should
+                    # be given if it doesn't exist:
+                    if no_acquire_flag and \
+                       hasattr(object, 'aq_base') and \
+                       not hasattr(object, '__bobo_traverse__'):
+
+                        if (object.__parent__ is not
+                                aq_inner(object).__parent__):
+                            from webdav.NullResource import NullResource
```

Comment:
Could this import be moved to the top?

Reply:
It cannot due to circular imports. I tried.